### ISSUE TITLE
MONAI-2D training settings

### ISSUE THREAD

**Issue by Maxsim111** [Cardio-AI/memorization-ldm#2]
**Very good job!**
I am attempting to reproduce your work, and I have been using the MONAI-2D training code provided through your shared link. I have observed that, as the training progresses, the model tends to memorize more training samples, which is consistent with your findings.

However, I am still facing two main issues:

1.The number of memorized samples is relatively small — around 1000 out of 10,000.
2.The growth trend is very slow — even after extending the training to 6000 epochs, I have not observed the rapid increase shown in your paper.
3.The FID of my trained MONAI-2D model remains quite high. This leads my advisor and me to suspect that my MONAI training configuration might not be correctly set up, resulting in a lower-quality model and consequently weaker memorization behavior.

To achieve high-quality MONAI-2D training for our subsequent experiments (which aim to reduce model memorization), I would greatly appreciate it if you could share the detailed steps and parameter settings you used for training MONAI-2D in your work.

I have attached some of my configuration files for your reference.
Thank you very much for your time and help!
#################Stage1#####################
seed=42
run_dir="aekl_v1"
batch_size=107
n_epochs=2100
adv_start=5
eval_freq=10
num_workers=24
experiment="AEKL"

**aekl.yaml**
stage1:
  base_lr: 0.00005
  disc_lr: 0.0001
  perceptual_weight: 0.002
  adv_weight: 0.005
  kl_weight: 0.00000001
  params:
    spatial_dims: 2
    in_channels: 1
    out_channels: 1
    num_channels: [64, 128, 128, 128]
    latent_channels: 1
    num_res_blocks: 2
    attention_levels: [False, False, False, False]
    with_encoder_nonlocal_attn: False
    with_decoder_nonlocal_attn: False

discriminator:
  params:
    spatial_dims: 2
    num_channels: 64
    num_layers_d: 3
    in_channels: 1
    out_channels: 1

perceptual_network:
  params:
    spatial_dims: 2
    network_type: "squeeze"
####################ldm######################
seed=2
run_dir="aekl_v1_ldm_v1"
scale_factor=0.3
batch_size=150
n_epochs=3000
eval_freq=10
num_workers=32
extended_report=0
experiment="LDM"

**ldm.yaml**
ldm:
  base_lr: 0.000025
  params:
    spatial_dims: 2
    in_channels: 1
    out_channels: 1
    num_res_blocks: 2
    num_channels: [256, 512, 768]
    attention_levels: [False, True, True]
    with_conditioning: True
    cross_attention_dim: 1024
    num_head_channels: [0, 512, 768]
  scheduler:
    schedule: "linear_beta"
    num_train_timesteps: 1000
    beta_start: 0.0015
    beta_end: 0.0205
    prediction_type: "v_prediction"

**Comment by salmandar** [comment]
I was using latent_channels=3 in the autoencoder training, and consequently, in_channels and out_channels were also equal to 3 in LDM training. How do the synthesized images look? Are they good-quality images?

Also, I trained AE for 400 epochs and a batch size of 12. I don't know how much this should matter, though.

**Comment by Maxsim111** [comment]
<img width="512" height="512" alt="Image" src="https://github.com/user-attachments/assets/586c3290-e214-4ebc-b1e4-307ee5c43ad5" />
<img width="512" height="512" alt="Image" src="https://github.com/user-attachments/assets/b388c2a6-a4c2-424a-b757-c1f21130b1d6" />

> I was using latent_channels=3 in the autoencoder training, and consequently, in_channels and out_channels were also equal to 3 in LDM training. How do the synthesized images look? Are they good-quality images?
> 
> Also, I trained AE for 400 epochs and a batch size of 12. I don't know how much this should matter, though.

Here are the samples generated at epoch 6000.I used InceptionV3 to compute FID, and the result was 19. Their quality is still not ideal. I have since increased latent_channels to 8 yesterday and the model is currently training. I will retrain the VAE following your configuration next. Did you modify any other hyperparameters or settings in addition to those you shared?
Thankyou very much！

**Comment by salmandar** [comment]
It seems like these are the only parameters. These images look blurry compared to the one I got after training my model. Also, if you look at the text ('L'), it's not properly synthesized. How do your reconstructions of the autoencoder look if you feed training samples to the autoencoder and reconstruct? Maybe you're also overtraining the autoencoder?

**Comment by Maxsim111** [comment]
> It seems like these are the only parameters. These images look blurry compared to the one I got after training my model. Also, if you look at the text ('L'), it's not properly synthesized. How do your reconstructions of the autoencoder look if you feed training samples to the autoencoder and reconstruct? Maybe you're also overtraining the autoencoder?

Hello, I trained a VAE with the above configuration where latent_channel = 1, and the reconstructed images turned out to be of very poor quality. Later, I modified the training setup by increasing the number of latent channels and enabling attention layers, which produced a new set of VAE reconstructions：
##########new stage1##############
stage1:
  base_lr: 0.00005
  disc_lr: 0.0001
  perceptual_weight: 0.001
  adv_weight: 0.001
  kl_weight: 0.00000001
  params:
    spatial_dims: 2
    in_channels: 1
    out_channels: 1
    num_channels: [128, 256, 256]
    latent_channels: 8
    num_res_blocks: 2
    attention_levels: [False, False, True]
    with_encoder_nonlocal_attn: False
    with_decoder_nonlocal_attn: False
#####################################
The results are much better than before, but still not entirely satisfactory. Do you think this level of reconstruction quality is sufficient to ensure that the second-stage experiments can proceed smoothly?

<img width="512" height="512" alt="Image" src="https://github.com/user-attachments/assets/8dadc871-c004-4f37-aafd-5c934004115c" />

<img width="512" height="512" alt="Image" src="https://github.com/user-attachments/assets/440be3d2-3e25-4f9b-b5ff-3dccf3c03432" />

**Comment by salmandar** [comment]
Actually, these images look poorer compared to what I have. For how many epochs have you trained the model (and as mentioned before, I used 3 latent channels)?

**Comment by Maxsim111** [comment]
> Actually, these images look poorer compared to what I have. For how many epochs have you trained the model (and as mentioned before, I used 3 latent channels)?
To maintain consistency with the training configuration of the LDM, I trained the VAE for a large number of epochs (2000 epochs) with the latent channel set to 1, which resulted in the poorly performing VAE mentioned above. Subsequently, I adjusted the batch size to 20, reduced the training to 250 epochs, and increased the latent channel  to 8, which produced the second set of images with slightly improved quality.I dont know if the second VAE is good enough to train the ldm.So I try to use VAE from Stable Diffusion-1.4 to train the ldm,and the code is running now.

**Comment by salmandar** [comment]
I can share my trained AE model with you.  Can you mail me or share with me your email ID?

**Comment by Maxsim111** [comment]
> I can share my trained AE model with you. Can you mail me or share with me your email ID?

Sure！My email is  25213060030@m.fudan.edu.cn.In the past few weeks, i tried to train AEKL and LDM with training settings you shared with me above(latent_channels=3,epoch=3000,batch_size=150...).However, the final training image memorizad was just 1600.What's more, the MONAI-2D code loads the "report" when training ldm, so it is trained conditioned.But your paper was unconditioned ldm,so did you remove it before training?My ldm before was trained conditioned , and the sampling was unconditioned(using None prompt), which cannot memorize training data, so I tried to sample with prompt(the labels),and Nmem increased(from 1600 to 2600) .I want to know if report should be removed when training.Thanks very much!

**Comment by salmandar** [comment]
I have mailed you the trained AE model and the code that I used for training.
